packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -224,6 +230,25 @@ export class EpochTracker implements IPersistedFileCache {
             }
             await this.checkForEpochError(error, epochFromResponse, fetchType);
             throw error;
+        }).catch((error) => {
+            // If the error is about location redirection, then we need to generate new resolved url with correct
+            // location info.
+            if (isFluidError(error) && error.errorType === DriverErrorType.locationRedirection) {
+                const redirectLocation = error.getTelemetryProperties().redirectLocation?.toString();
+                assert(redirectLocation !== undefined, "locationRedirection error should contain redirectLocation");
+                const patchedResolvedUrl = patchOdspResolvedUrl(
```

Comment:
I'd rename this to `redirectUrl` to match the parameter, esp it will be logged that way.  For a second I thought it was going to conflict with `redirectionLocation` telemetry prop but looks ok.